Make sure we type the cast properly.

http://bugs.gentoo.org/show_bug.cgi?id=63773

Pingus uses assert() but never includes assert.h, boo !

http://bugs.gentoo.org/show_bug.cgi?id=28281

--- pingus-0.6.0/src/story_screen.cxx
+++ pingus-0.6.0/src/story_screen.cxx
@@ -142,7 +142,7 @@
 
   if (!page_displayed_completly)
     {
-      unsigned int len = static_cast<unsigned int>(20.0f * time_passed);
+      size_t len = static_cast<size_t>(20.0f * time_passed);
       display_text = current_page.text.substr(0, Math::min(current_page.text.length(), len));
 
       if (current_page.text.length() < len)
--- pingus-0.6.0/src/blitter.cxx
+++ pingus-0.6.0/src/blitter.cxx
@@ -17,6 +17,7 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <assert.h>
 #include <config.h>
 #include <stdio.h>
 #include "pingus_error.hxx"
--- pingus-0.6.0/src/col_map.cxx
+++ pingus-0.6.0/src/col_map.cxx
@@ -17,6 +17,7 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <assert.h>
 #include <iostream>
 #include <ClanLib/Display/SurfaceProviders/canvas.h>
 #include "gui/graphic_context.hxx"
--- pingus-0.6.0/src/console.cxx
+++ pingus-0.6.0/src/console.cxx
@@ -17,6 +17,7 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <assert.h>
 #include <config.h>
 #include <ClanLib/Display/Display/display.h>
 #include <ClanLib/Display/Font/font.h>
--- pingus-0.6.0/src/demo_recorder.cxx
+++ pingus-0.6.0/src/demo_recorder.cxx
@@ -17,6 +17,7 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <assert.h>
 #include <time.h>
 #include <iostream>
 #include "system.hxx"
--- pingus-0.6.0/src/editor/generic_property_frame.cxx
+++ pingus-0.6.0/src/editor/generic_property_frame.cxx
@@ -17,6 +17,7 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <assert.h>
 #include <iostream>
 #include <ClanLib/GUI/label.h>
 #include <ClanLib/GUI/inputbox.h>
--- pingus-0.6.0/src/gui/root_gui_manager.cxx
+++ pingus-0.6.0/src/gui/root_gui_manager.cxx
@@ -17,6 +17,7 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <assert.h>
 #include "root_gui_manager.hxx"
 #include "../input/controller.hxx"
 
--- pingus-0.6.0/src/pingu_holder.cxx
+++ pingus-0.6.0/src/pingu_holder.cxx
@@ -17,6 +17,7 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <assert.h>
 #include <iostream>
 #include "pingu_holder.hxx"
 #include "pingu.hxx"
--- pingus-0.6.0/src/savegame_manager.cxx
+++ pingus-0.6.0/src/savegame_manager.cxx
@@ -17,6 +17,7 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <assert.h>
 #include <iostream>
 #include <fstream>
 #include "system.hxx"
--- pingus-0.6.0/src/worldmap/dot.cxx
+++ pingus-0.6.0/src/worldmap/dot.cxx
@@ -17,6 +17,7 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <assert.h>
 #include <iostream>
 #include "../xml_file_reader.hxx"
 #include "dot.hxx"
--- pingus-0.6.0/src/worldmap/graph.hxx
+++ pingus-0.6.0/src/worldmap/graph.hxx
@@ -20,6 +20,8 @@
 #ifndef HEADER_GRAPH_HXX
 #define HEADER_GRAPH_HXX
 
+#include <assert.h>
 #include <algorithm>
 #include <vector>
+#include <iostream>
 
diff -ur pingus-0.6.0-orig/src/worldmap/sprite_drawable.hxx pingus-0.6.0/src/worldmap/sprite_drawable.hxx
--- pingus-0.6.0-orig/src/worldmap/sprite_drawable.hxx	2005-09-11 21:46:11.000000000 -0400
+++ pingus-0.6.0/src/worldmap/sprite_drawable.hxx	2005-09-11 21:49:46.000000000 -0400
@@ -46,7 +46,7 @@
 
 private:
   SpriteDrawable (const SpriteDrawable&);
-  SpriteDrawable operator= (const SpriteDrawable&);
+  SpriteDrawable& operator= (const SpriteDrawable&);
 };
 
 } // namespace WorldMapNS
